lua/orgmode/utils/init.lua

```diff
@@ -559,9 +561,14 @@ function utils.open_tmp_org_window(height, split_mode, on_close)
   return function()
     vim.api.nvim_create_augroup('OrgTmpWindow', { clear = true })
     local prev_winnr = vim.api.nvim_buf_get_var(0, 'org_prev_window')
-    vim.api.nvim_win_close(0, true)
-    if prev_winnr and vim.api.nvim_win_is_valid(prev_winnr) then
-      vim.api.nvim_set_current_win(prev_winnr)
+    if prev_winnr ~= vim.api.nvim_get_current_win() then
+      vim.api.nvim_win_close(0, true)
+      if prev_winnr and vim.api.nvim_win_is_valid(prev_winnr) then
+        vim.api.nvim_set_current_win(prev_winnr)
+      end
+    else
+      local prev_bufnr = vim.api.nvim_buf_get_var(0, 'org_prev_buffer')
+      vim.api.nvim_set_current_buf(prev_bufnr)
```

Comment:
Probably best to also run the checks on `bufnr`.

```
if prev_bufnr and vim.api.nvim_buf_is_valid(prev_bufnr) ...
```

Reply:
btw, what should happen if the `bufnr` is invalid? With `winnr` the window gets closed in any case.

Reply:
Good point(s). I suppose if the `prev_bufnr` is invalid, we should simply unload the capture buffer and vim will automatically switch to some other buffer?